### PR TITLE
evaluator rebulit

### PR DESCRIPTION
**Pull request by jalynnliu**
this version of evaluator.py re-construct the whole code structure without any actual logic change or improvement on train step. notice that this version haven't been tested with multi-block method.

### CODE REVIEW

**Review by bluebread — APPROVED** [review]
1. 加 dropout
2. 改 evalute 接口和存储模型的方式
3. 添加 learning_rate 种类

**Comment by bluebread** [comment]
當訓練結果很差時，廢棄網絡時，返回 -1。
